### ISSUE TITLE
fm311 fm351 Expression analysis failed on: Expr -> Designator -> DataRef -> ArrayElement

### ISSUE THREAD

**Issue by schweitzpgi** [flang-compiler/f18-llvm-project#181]
fcvs tests fm311 and fm351

```
flang -c   /proj/ta/tests/fcvs/src/fm351.f
/proj/ta/tests/fcvs/src/fm351.f:52:31: error: Internal error: Expression analysis failed on: Expr -> Designator -> DataRef -> ArrayElement
| DataRef -> Name = 'iadn11'
| SectionSubscript -> Integer -> Expr -> Designator -> DataRef -> Name = 'idon04'

        IFOS02(IDON04,IDON05) = IADN11(IDON04) / IADN11(IDON05)
                                ^^^^^^^^^^^^^^
```

**Comment by klausler** [comment]
Expression analysis of statement function bodies takes place before implicit typing rules have been applied to symbols in the containing specification part.  A fix is on its way that defers that analysis until the specification part is finished.

**Comment by klausler** [comment]
Fix is in llvm-project/master; please verify after next pull & rebase and close this report.

**Comment by schweitzpgi** [comment]
This issue is fixed. Closing. (New issues can be filed for additional exposed problems.)

---------------------------------- executing test fm351
./fm351.out > fm351.log
IEEE arithmetic exceptions signaled: INEXACTdiff fm351.log .//ref/fm351.log
37c37
<         25       FAIL    -0.71919E+03    0.84234E+02
---
>         25       PASS
43,44c43,44
<                         1 TESTS FAILED
<                        24 TESTS PASSED
---
>                         0 TESTS FAILED
>                        25 TESTS PASSED
